openedx/features/partners/tests/test_g2a_registration.py

```diff
@@ -0,0 +1,175 @@
+
+from lms.djangoapps.onboarding.models import Organization, UserExtendedProfile
+from openedx.core.lib.api.test_utils import ApiTestCase
+from openedx.features.partners.tests.factories import FocusAreaFactory, OrganizationFactory, PartnerFactory
+
+
+@ddt
+class RegistrationViewTest(ApiTestCase):
+    """
+    Includes test cases for openedx partner feature for registration
+    """
+
+    NAME = 'bob james'
+    COUNTRY = 'Pakistan'
+    ORGANIZATION = 'arbisoft'
+    EMAIL = 'bob@example.com'
+    USERNAME = 'bob'
+    PASSWORD = 'Test@12345'
+    TERMS_OF_SERVICE = True
+
+    PARTNER = 'give2asia'
+
+    def setUp(self):
+        self.partner = PartnerFactory.create(slug=self.PARTNER, label=self.ORGANIZATION)
+        self.registration_url = reverse('partner_register', args=[self.PARTNER])
+
+    def test_put_not_allowed(self):
+        """
+        API should not accept http put method
+        :return : None
+        """
+        response = self.client.put(self.registration_url)
+        self.assertHttpMethodNotAllowed(response)
+
+    def test_delete_not_allowed(self):
+        """
+        API should not accept http delete method
+        :return : None
+        """
+        response = self.client.delete(self.registration_url)
+        self.assertHttpMethodNotAllowed(response)
+
+    def test_create_new_partner_user_with_organization(self):
+        """
+        Test user is not first learner if organization already exists
+        Create an organization and then try to register.
+        :return : None
+        """
+        OrganizationFactory(label=self.ORGANIZATION)
+
+        response = self.client.post(self.registration_url, {
+            'name': self.NAME,
+            'organization_name': self.ORGANIZATION,
+            'country': self.COUNTRY,
+            'email': self.EMAIL,
+            'username': self.USERNAME,
+            'password': self.PASSWORD,
+            'terms_of_service': self.TERMS_OF_SERVICE
+        })
+        self.assertHttpOK(response)
+
+        # get inserted data from the database
+        user = User.objects.filter(username=self.USERNAME).first()
+        organization = Organization.objects.filter(label__iexact=self.ORGANIZATION).first()
+        extended_profile = UserExtendedProfile.objects.filter(user=user).first()
+
+        # verify if user is registered and data is stored int he database
+        self.assertNotEqual(user, None)
+        self.assertEqual(self.USERNAME, user.username)
+        self.assertEqual(self.EMAIL, user.email)
+        self.assertEqual(self.NAME.split(' ', 1)[0], user.first_name)
+        self.assertEqual(self.NAME.split(' ', 1)[1], user.last_name)
+        self.assertEqual(True, user.is_active)
+        self.assertEqual(organization.label, organization.label)
+        # in case organization already exists make sure that user is not first learner
+        self.assertEqual(extended_profile.is_first_learner, False)
+        self.assertHttpOK(response)
```

Comment:
Lets use:

`assertIsNotNone()` instead of comparing to `None`
`assertTrue()` instead of comparing to `True`
`assertFalse()` instead of comparing to `False`

Please update all occurrences

Reply:
done